src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -39,60 +38,56 @@ public void editPerson_usingAccelerator() {
         //Get a reference to the card displaying Alice's details
         PersonCardHandle alicePersonCard = personListPanel.getPersonCardHandle(td.alice);
 
-        //Edit Alice to change to new values
-        personListPanel.clickOnPerson(td.alice);
-        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
-        editPersonDialog.enterNewValues(newAlice);
-        editPersonDialog.pressEnter();
+        personListPanel.rightClickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =
+                personListPanel.clickOnContextMenu(PersonListPanelHandle.ContextMenuChoice.EDIT);
+        assertTrue(editPersonDialog.isShowingPerson(td.alice));
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
 
         //Confirm pending state correctness
         assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
 
         //Confirm the right card is selected after the edit
-        assertEquals(alicePersonCard, newAlice);
+        assertMatching(alicePersonCard, newAlice);
         assertTrue(personListPanel.isSelected(newAlice));
 
-
         //Confirm right values are displayed after grace period is over
         sleepForGracePeriod();
-        assertEquals(alicePersonCard, newAlice);
+        assertMatching(alicePersonCard, newAlice);
 
+        //Confirm cancel operation does not cancel the edit after the grace period.
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         //Confirm the underlying person object has the right values
-        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
-
+        assertMatching(alicePersonCard, newAlice);
 
         //Confirm again after the next sync
         sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
         //Confirm other cards are unaffected.
-        personListPanel.clickOnListView();
         assertTrue(personListPanel.isListMatching(1, td.benson, td.charlie, td.dan, td.elizabeth));
 
         //Confirm status bar is updated correctly
         assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] completed successfully.");
-    }
 
-    @Test
-    public void editPerson_usingContextMenu() {
-        personListPanel.rightClickOnPerson(td.alice);
-        EditPersonDialogHandle editPersonDialog = personListPanel.clickOnContextMenu(
-                PersonListPanelHandle.ContextMenuChoice.EDIT);
-        assertTrue(editPersonDialog.isValidEditDialog());
     }
 
+    //TODO: This maybe should not be here. A separate class to test the button does what it suppose to do?
```

Comment:
My idea was to separate the button and accelerator test to its own separate classes. For example, the method here is just to test if the button open the edit dialog box. So the main thing that is testing is whether the button works but not on editing person.

And on editing person test, the test should be testing on the editing person process and not whether if the button opens the dialog box.

I guess it is a matter of definition here. I feel that it will help in preventing a bloating test class(testing too many things).